src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/get/TransportGetTransformsAction.kt

```diff
@@ -0,0 +1,85 @@
+package com.amazon.opendistroforelasticsearch.indexmanagement.transform.action.get
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.IndexManagementPlugin.Companion.INDEX_MANAGEMENT_INDEX
+import com.amazon.opendistroforelasticsearch.indexmanagement.elasticapi.parseWithType
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.model.Transform
+import org.elasticsearch.ElasticsearchStatusException
+import org.elasticsearch.ExceptionsHelper
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.search.SearchRequest
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.action.support.ActionFilters
+import org.elasticsearch.action.support.HandledTransportAction
+import org.elasticsearch.client.Client
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentHelper
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.index.query.BoolQueryBuilder
+import org.elasticsearch.index.query.ExistsQueryBuilder
+import org.elasticsearch.index.query.WildcardQueryBuilder
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.search.SearchModule
+import org.elasticsearch.search.builder.SearchSourceBuilder
+import org.elasticsearch.search.sort.SortOrder
+import org.elasticsearch.tasks.Task
+import org.elasticsearch.transport.TransportService
+
+class TransportGetTransformsAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters,
+    val xContentRegistry: NamedXContentRegistry
+) : HandledTransportAction<GetTransformsRequest, GetTransformsResponse> (
+    GetTransformsAction.NAME, transportService, actionFilters, ::GetTransformsRequest
+) {
+
+    override fun doExecute(task: Task, request: GetTransformsRequest, listener: ActionListener<GetTransformsResponse>) {
```

Comment:
This probably is same as `rollup` logic can you instead create an util method to hold this and reuse this in both rollups and here?

Reply:
Wrote common logic and applied it to Transform class. Haven't applied logic to Rollup class yet.